### DISCUSSION TITLE
issue: fields that use Controller or useController with its own defaultValue don't reset when resetField() is called after unmounting

### DISCUSSION THREAD

**Discussion by andre-kingsmen** [react-hook-form/react-hook-form#10572] [Q&A]
### Version Number

7.45.0

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/dreamy-taussig-zhjn4k

### Steps to reproduce

1. type in something to the input, ex: "asd"
2. click on "show results" (which will "unrender" / "unmount" the form)
3. click on "edit" (which will render the form again)
4. click on "reset field" (calls `resetField("firstName")`)

### Expected behaviour

**Results:**
- the value in the input is still on "asd" when it should be set to undefined
- though the `useForm()` has the correct values, check the logs (which shows `watch("firstName")`, correctly showing undefined), although it should technically be the value of the defaultValue of the <Controller> - which is another issue i think

**Notes:**
- when defaultValue is set with something, ex: `useForm({ defaultValue: "" })`, it does work properly
- `setValue("firstName", "somethin")` does work, click on "set value" to see

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please provide defaultValues: `useForm<FormInputs>({ defaultValues: {} });` resetField tries to get what's in the defautlValues but it's undefined with what you have provided.

**Reply by andre-kingsmen** [reply]
yes adding defaultValues fixes it, but the default behavior is that when u dont have defaultValues in `useForm()` it uses the defaultValue of the Controller - which is how it works before the "underender" step, just not after it. rn behavior is not consistent.

and also, useForm() with empty defaultValues should have actually made it `undefined` right? its the default behavior - but its not updating it to that either

**Comment by kimyooeco** [comment]
Is it correct that it hasn't been resolved?
